File: ai.py
import asyncio
import aiohttp
from telethon import TelegramClient, events, Button
from telethon.tl.types import InputBotInlineResult, InputWebDocument

# مقادیر API خود را اینجا وارد کنید
api_id = '18377832'
api_hash = 'ed8556c450c6d0fd68912423325dd09c'
bot_token = '7000850548:AAEZ1JJfZ6QhNwe8Z9qsrGzd9hHZBp_iIno'
admin_id = 6856915102

# ایجاد نمونه ربات
client = TelegramClient('bot_session', api_id, api_hash).start(bot_token=bot_token)


async def fetch_api(url, json_data, headers):
    """ارسال درخواست به API هوش مصنوعی"""
    async with aiohttp.ClientSession() as session:
        async with session.post(url, json=json_data, headers=headers) as response:
            # دریافت محتوای پاسخ به صورت متنی
            text_response = await response.text()
            return text_response


async def chat_with_ai(query, user_id):
    """ارسال پیام کاربر به API و دریافت پاسخ"""
    url = "https://api.binjie.fun/api/generateStream"
    headers = {
        "authority": "api.binjie.fun",
        "accept": "application/json, text/plain, */*",
        "accept-language": "en-US,en;q=0.9",
        "origin": "https://chat18.aichatos.xyz",
        "referer": "https://chat18.aichatos.xyz/",
        "user-agent": "Mozilla/5.0 (Windows NT 6.3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36",
        "Content-Type": "application/json"
    }
    data = {
        "prompt": query,
        "userId": str(user_id),
        "network": True,
        "system": "",
        "withoutContext": False,
        "stream": False
    }
    response_text = await fetch_api(url, json_data=data, headers=headers)
    return response_text if response_text else "متاسفم، پاسخ مناسبی دریافت نشد."


@client.on(events.NewMessage(pattern='/start'))
async def start_handler(event):

    user = await event.get_sender()  # دریافت اطلاعات کاربر
    message = (
        f"🚀 **یک کاربر جدید ربات را استارت کرد!**\n\n"
        f"👤 نام: {user.first_name}\n"
        f"🆔 آیدی: `{user.id}`"
    )


    
    """ارسال پیام راهنما هنگام استارت ربات"""
    start_text = (
        "🤖 **به ربات هوش مصنوعی خوش آمدید!**\n\n"
        "📌 **نحوه استفاده:**\n"
        "1️⃣ پیام خود را ارسال کنید.\n"
        "2️⃣ ربات پاسخ شما را می‌دهد.\n"
        "🚀 **برای شروع، پیام خود را ارسال کنید!**"
    )

    # ارسال پیام با دکمه اینلاین
    await event.reply(
        start_text,
        buttons=[Button.url("✍️ نویسنده: احسان فضلی", "https://t.me/abj0o")],
        link_preview=False
    )
    
    if admin_id:
        try:
            await bot.send_message(admin_id, message)
        except Exception as e:
            print(f"⚠️ ارسال پیام به ادمین با خطا مواجه شد: {e}")
    


@client.on(events.NewMessage)
async def handler(event):
    """دریافت پیام کاربر و پردازش آن"""
    user_message = event.message.message
    user_id = event.sender_id
    chat_id = event.chat_id

    # ارسال واکنش به پیام کاربر با ایموجی 👍
    try:
        async with client.action(chat_id, "typing"):
    # دریافت پاسخ از API هوش مصنوعی
            response_text = await chat_with_ai(user_message, user_id)

    # ارسال پاسخ به عنوان ریپلای همراه با دکمه اینلاین 🥰
            await event.reply(
                response_text,
                buttons=[Button.inline("🥰", b"love")]
            )
    except Exception as e:
        print(f"Error occurred: {e}")
@client.on(events.NewMessage)
async def forward_to_admin(event):
    """فوروارد تمامی پیام‌ها به ادمین"""
    sender = await event.get_sender()
    if sender.id != admin_id:  # جلوگیری از ارسال پیام‌های ادمین به خودش
        try:
            await bot.forward_messages(admin_id, event.message)


@client.on(events.CallbackQuery(data=b"love"))
async def love_callback(event):
    """پاسخ به کلیک روی دکمه 🥰"""
    await event.answer("❤️ ممنون از من استفاده می کنید امید وار کمک رسان خوبی به شما باشم !", alert=True)


# شروع ربات
client.start()
print("ربات فعال است...")
client.run_until_disconnected()

File: ai.js
const express = require("express");
const { Telegraf, Markup } = require("telegraf");
const axios = require("axios");
const bodyParser = require("body-parser");

const app = express();
app.use(bodyParser.json());

// مقادیر خود را جایگزین کنید
const botToken = "7000850548:AAH5oF7R6AYdDp5RJCaPiK2-bx5EwygoaG4";
const adminId = 6856915102;

const bot = new Telegraf(botToken);

// Webhook endpoint
app.use(bot.webhookCallback("/bot"));

// تنظیم وبهوک (تنها در اولین اجرای deploy روی Vercel یا به‌صورت دستی با curl)
bot.telegram.setWebhook("https://ai-bot-ehsan.vercel.app/bot");

// تابع تماس با API هوش مصنوعی
async function chatWithAI(query, userId) {
  try {
    const url = "https://api.binjie.fun/api/generateStream";
    const headers = {
      "authority": "api.binjie.fun",
      "accept": "application/json, text/plain, */*",
      "accept-language": "en-US,en;q=0.9",
      "origin": "https://chat18.aichatos.xyz",
      "referer": "https://chat18.aichatos.xyz/",
      "user-agent": "Mozilla/5.0",
      "Content-Type": "application/json"
    };

    const data = {
      prompt: query,
      userId: String(userId),
      network: true,
      system: "",
      withoutContext: false,
      stream: false
    };

    const response = await axios.post(url, data, { headers });
    return response.data || "متاسفم، پاسخ مناسبی دریافت نشد.";
  } catch (error) {
    console.error("❌ خطا در ارتباط با API:", error);
    return "❌ خطایی رخ داد، لطفاً مجدداً تلاش کنید.";
  }
}

// /start command
bot.start(async (ctx) => {
  const user = ctx.from;
  const startText = `🤖 **به ربات هوش مصنوعی خوش آمدید!**\n\n`
    + `📌 **نحوه استفاده:**\n`
    + `1️⃣ پیام خود را ارسال کنید.\n`
    + `2️⃣ ربات پاسخ شما را می‌دهد.\n`
    + `🚀 **برای شروع، پیام خود را ارسال کنید!**`;

  await ctx.reply(startText, {
    parse_mode: "Markdown",
    ...Markup.inlineKeyboard([
      Markup.button.url("✍️ نویسنده: احسان فضلی", "https://t.me/abj0o"),
    ])
  });

  const adminMessage = `🚀 **یک کاربر جدید ربات را استارت کرد!**\n\n`
    + `👤 نام: ${user.first_name}\n`
    + `🆔 آیدی: \`${user.id}\``;

  if (adminId) {
    try {
      await bot.telegram.sendMessage(adminId, adminMessage, { parse_mode: "Markdown" });
    } catch (error) {
      console.error("⚠️ خطا در ارسال پیام به ادمین:", error);
    }
  }
});

// هندل پیام‌ها
bot.on("text", async (ctx) => {
  const userMessage = ctx.message.text;
  const userId = ctx.from.id;

  try {
    await ctx.sendChatAction("typing");
    const responseText = await chatWithAI(userMessage, userId);
    await ctx.reply(responseText, {
      ...Markup.inlineKeyboard([
        Markup.button.callback("🥰", "love"),
      ])
    });
  } catch (error) {
    console.error("خطا در پردازش پیام:", error);
  }
});

// پاسخ به دکمه
bot.action("love", async (ctx) => {
  await ctx.answerCbQuery("❤️ ممنون از من استفاده می‌کنید، امیدوارم کمک‌رسان خوبی باشم!", { show_alert: true });
});

module.exports = app;
